### PR TITLE
{bio}[foss/2015b] BBMap 35.82 (REVIEW)

### PR DESCRIPTION
**Pull request by nathanhaigh**

### CODE REVIEW

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(name)s_%(version)s.tar.gz' ]
+
+prebuildopts = 'cd jni && '
+
+os_type = platform.system()
+if os_type == 'Linux':
+    buildopts = '-f makefile.linux'
+elif os_type == 'Darwin':
+    buildopts = '-f makefile.osx'
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: things like this should be avoided in easyconfigs

please check if there's a constant available for this (see --avail-easyconfig-constants); if not, we'll add the one you require here

The if/else should also be avoided, maybe use a dictionary that maps value of constant to the label to use?

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+files_to_copy = ['./*']
+
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: no need for the ./ ?

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: This should work:

```python
suff = {'Darwin': 'osx', 'Linux': 'linux'}[OS_TYPE]
buildopts = "-f makefile.%s" % suff

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import platform
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: eek, please remove :)

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_74.eb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import platform
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_74.eb]: please remove?

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_74.eb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+files_to_copy = ['*']
+
+sanity_check_paths = {
+    'files': ['bbmap.sh'],
+    'dirs': []
+}
+
+modextrapaths = {'PATH': ''}
+
+modloadmsg = "For improved speed, add 'usejni=t' to the command line of %(name)s tools which support the use of the compiled jni C code."
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_74.eb]: keep lines shorter than 120 characters please

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+files_to_copy = ['*']
+
+sanity_check_paths = {
+    'files': ['bbmap.sh'],
+    'dirs': []
+}
+
+modextrapaths = {'PATH': ''}
+
+modloadmsg = "For improved speed, add 'usejni=t' to the command line of %(name)s tools which support the use of the compiled jni C code."
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: keep lines shorter than 120 characters please

**Comment by boegel** [comment]
missing Java easyconfig is being added in #2545

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+sanity_check_paths = {
+    'files': ['bbmap.sh'],
+    'dirs': []
+}
+
+modextrapaths = {'PATH': ''}
+
+modloadmsg = "For improved speed, add 'usejni=t' to the command line of %(name)s tools which support the use of the
+  compiled jni C code."
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: you'll need to use triple quotes, or

```python
modloadmsg = ""For improved speed, add 'usejni=t' to the command line of %(name)s tools "
modloadmsg += "which support the use of the compiled jni C code."
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/d4985257ea93a86ce8c8 for a full test report.

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+
+java = 'Java'
+javaver = '1.8.0_66'
+versionsuffix = '-%s-%s' % (java, javaver)
+dependencies = [(java, javaver, '', True)]
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(name)s_%(version)s.tar.gz' ]
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: @nathanhaigh: last remark: please move this up, below `toolchain`, to be consistent with other easyconfigs

File: easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+name = 'BBMap'
+version = '35.82'
+
+homepage = 'https://sourceforge.net/projects/bbmap/'
+description = """BBMap short read aligner, and other bioinformatic tools."""
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(name)s_%(version)s.tar.gz' ]
+
+toolchain = {'name': 'foss', 'version': '2015b'}
```
Comment (boegel) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: @nathanhaigh: at the risk of coming across as an annoying brat: `source_urls` and `sources` go *below* `toolchain` usually...

Comment (nathanhaigh) [inline easybuild/easyconfigs/b/BBMap/BBMap-35.82-foss-2015b-Java-1.8.0_66.eb]: Nope problem! Anyway, I'm the annoying one for not following instruction!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/1004108877e16919ec23 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @nathanhaigh!

**Pull request merged**